frontend/src/pulldasher/filtered-pulls-state.ts

```diff
@@ -0,0 +1,22 @@
+import { useState } from 'react';
+import { usePullsState } from './pulls-state';
+import { Pull } from  '../pull';
+
+export type FilterFunction = (pull: Pull) => boolean;
+export type FilterFunctionSetter = (filter: FilterFunction|null) => void;
+export interface FilterProps {
+   filter: FilterFunction
+}
+
+const defaultFilter = (pull: Pull) => !!pull;
+
+/**
+ * Wrapper around usePullsState that provides filtering
+ */
+export function useFilteredPullsState(): [Pull[], FilterFunctionSetter] {
+   const [{filter}, setFilter] = useState<FilterProps>({filter: defaultFilter});
+   return [
+      usePullsState().filter(filter),
+      (filter: FilterFunction) => setFilter({filter: filter || defaultFilter}),
+   ];
+}
```

Comment:
Why the object, instead of storing `filter` directly in the `useState`?

Reply:
Turns out, you can't store a function directly as state. `useState()` and the `setState` function both do a different thing when passed a function.

See: https://medium.com/swlh/how-to-store-a-function-with-the-usestate-hook-in-react-8a88dd4eede1